src/sentry/integrations/opsgenie/actions/form.py

```diff
@@ -0,0 +1,84 @@
+# mypy: ignore-errors
+from __future__ import annotations
+
+from typing import Any, Mapping
+
+from django import forms
+from django.utils.translation import gettext_lazy as _
+
+from sentry.services.hybrid_cloud.integration import integration_service
+from sentry.services.hybrid_cloud.integration.model import RpcOrganizationIntegration
+
+
+def _validate_int_field(field: str, cleaned_data: Mapping[str, Any]) -> int | None:
+    value_option = cleaned_data.get(field)
+    if value_option is None:
+        return None
+
+    try:
+        return int(value_option)
+    except ValueError:
+        raise forms.ValidationError(_(f"Invalid {field}"), code="invalid")
+
+
+class OpsgenieNotifyTeamForm(forms.Form):
+    """Used for notifying a specific team."""
+
+    account = forms.ChoiceField(choices=(), widget=forms.Select())
+    team = forms.ChoiceField(required=False, choices=(), widget=forms.Select())
```

Comment:
I know it shouldn't be a visible field, but I do think it'd help to save the `organization_integration_id` on the alert so that you don't have to iterate through all of them to find the associated `team_id`. Especially since they aren't enforced to be unique across organizations.

Hypothetical example:
- Sentry Org A installs `my-opsgenie` with Team `test1` and key `abc`
- Sentry Org B installs `my-opsgenie` with Team `test1` and key `def`
- When alerting Org A, we want to ensure we use the key `abc` not `def`, in case they have access logs or something associated with the keys, even if both my work.

Reply:
Since we don't have access to the org_integration until something is selected, I'm going to save the org_id so that we can get the org_integration associated with a specific org.